src/server-logger.ts

```diff
@@ -0,0 +1,257 @@
+import request from 'superagent';
+import stringify from 'safe-json-stringify';
+import cloneDeep from 'lodash.clonedeep';
+
+import { Logger } from './logger';
+import { IDetail, ILoggerConfig, LogLevel, ITrace, ILogBufferItem, ILogMessage } from './interfaces';
+import { calculateLogMessageSize } from './utils';
+
+const MAX_LOG_SIZE = 14500;
+const DEFAULT_UPLOAD_DEBOUNCE = 4000;
+
+export class ServerLogger {
+  private config: ILoggerConfig;
+  private isInitialized: boolean = false;
+  private logger: Logger;
+  private logBuffer: Array<ILogBufferItem> = [];
```

Comment:
I found it is a little more simple if we have an array of buffer items and each buffer item will be less than the max allowed payload size.